robot/robot.go

```diff
@@ -68,15 +72,35 @@ func (r *Robot) Move(steps int) error {
 	case West:
 		newPosition.X = r.Position.X - steps
 	}
-	if r.EnvMap.isObstacleAt(newPosition) {
-		return fmt.Errorf("The robot can't move %v steps because their are obstacles in the way.Here's the current map: %v", steps, mapWithRobot(r))
-	}
 	distance := Centimeters(float64(steps) * float64(r.EnvMap.SquareSize))
 	r.MoveModule.Move(distance, r.Tachometer)
 	r.Position = newPosition
 	return nil
 }
 
+func (r *Robot) canMoveToPosition(steps int) bool {
+	newPosition := r.Position
+	absSteps := int(math.Abs(float64(steps)))
+	singleStep := steps / absSteps
+	for i := 0; i < absSteps; i++ {
+		switch r.Position.Orientation {
```

Comment:
Did that (commit 5b935be6c2ac80a3606fabc11c440e69f4153d24).
The implementation as it is now is actually what I intended the first time. I just ran into some problems with the distance verification in unit tests. In short my robot would move until it hit an obstacle and then always only move the distance of a single step.
During refactoring I figured out what the problem was and now it's working! :-D